### ISSUE TITLE
Libc: remove 'clock()'

### ISSUE THREAD

**Issue by ehmry** [genodelabs/genode#3057]
> The clock() function determines the amount	of processor time used since
> the invocation of the calling process, measured in	CLOCKS_PER_SECs	of a
> second.

This procedure is just a shallow wrapper over `getrusage` which we cannot reasonably implement. Whats worse, `clock` is so inaccurate on FreeBSD its not even useful.

> ... the Single UNIX	Specification (``SUSv2'') requires
> CLOCKS_PER_SEC to be defined as one million.  FreeBSD does	not conform to
> this requirement; changing	the value would	introduce binary incompatibil-
> ity and one million is still inadequate on	modern processors.

```c
/* Frequency of the clock ticks reported by clock().  */
#define	CLOCKS_PER_SEC	128

```

**Comment by ehmry** [comment]
Commits at 50b7b3ec4d1920eb1e3731ccdf8e685022b70d63 and 7c03ed1beceaca3543a414754852666df7a4c78e.

**Comment by nfeske** [comment]
Just out of curiosity, can you share the anecdote that triggered you to remove it?

**Comment by ehmry** [comment]
Yes, I was measuring times in the 9P server using the Nim `cpuTim()` procedure, but `cpuTime()` was returning an odd number. This was because the FreeBSD implementation was calling the dummy `getrusage` and returning -1, and Nim was converting -1 to a floating point seconds value. I then reimplemented `clock()` to avoid `getrusage`, but then the result seemed uncomfortably imprecise. That was when I discovered that FreeBSD has kept the unit of time returned by `clock()` at precision of the original implementation for binary compatibility.

https://github.com/nim-lang/Nim/pull/9793
